### PR TITLE
[IMP] membership_withdrawal: allow to archive reasons

### PR DESCRIPTION
**Pull request by chienandalu**
cc @Tecnativa TT26867

### CODE REVIEW

**Review by pedrobaeza — COMMENTED** [review]
And the UI?

**Comment by chienandalu** [comment]
Well, when the field is present in the model you can always archive-unarchive selecting records in the tree view and going to *Action*

**Comment by pedrobaeza** [comment]
But on v12 the usual thing is the smart-button, and I'm not sure buttons appear for this version.

**Comment by chienandalu** [comment]
The thing is there's no proper form view defined for this model, only an editable tree

**Comment by pedrobaeza** [comment]
OK, maybe include the field there?

**Comment by chienandalu** [comment]
Mm, maybe a better improvement would be to add a search view for archived items
image

**Review by pedrobaeza — APPROVED** [review]

File: membership_withdrawal/views/membership_withdrawal_reason_view.xml [hunk 1]
```diff
@@ -16,10 +16,22 @@
     <field name="arch" type="xml">
         <tree string="Membership withdrawal reasons" editable="top">
             <field name="name"/>
+            <field name="active" readonly="1" />
```
Comment (astirpe) [inline membership_withdrawal/views/membership_withdrawal_reason_view.xml:19]: I think this solution is ok, although I would prefer to have a form view defined for this model and a toggle button, to be consistent with the standard V12 archival feature.

File: membership_withdrawal/views/membership_withdrawal_reason_view.xml [hunk 2]
```diff
@@ -16,10 +16,22 @@
...
     </field>
 </record>
 
+<record id="membership_withdrawal_reason_search_view" model="ir.ui.view">
+    <field name="model">membership.withdrawal_reason</field>
+    <field name="arch" type="xml">
+        <search>
+            <filter string="All" domain="['|', ('active','=', True), ('active','=', False)]" name="all_reasons"/>
+            <filter string="Active" domain="[('active','=', True)]" name="active_reasons"/>
+            <filter string="Archived" domain="[('active','=', False)]" name="archived_reasons"/>
```
Comment (astirpe) [inline membership_withdrawal/views/membership_withdrawal_reason_view.xml:29]: Only the "Archived" filter is needed, the other two filters are not strictly necessary and could be removed.

**Review by astirpe — APPROVED** [review]

File: membership_withdrawal/views/membership_withdrawal_reason_view.xml (hunk 2 above)
Comment (pedrobaeza) [reply membership_withdrawal/views/membership_withdrawal_reason_view.xml:29]: Although the `All` filter can be removed, the other is useful for allowing to see both together (active and non active). They should be anyway split with `<separator>` elements for not combining with other filter elements.

**Comment by pedrobaeza** [comment]
/ocabot merge minor

Please fw-port to v13/v14

**Pull request merged**